src/app/components/room/room.service.js

```diff
@@ -213,8 +213,28 @@
       this.publishingFromStart = config.publishingFromStart || false;
     }
 
-    function getRoom() {
-      return this.room;
+    function getRoomById(roomId) {
+      var deferred = $q.defer();
+      var that = this;
+
+      that.connect().then(function () {
+        that.getAvailableRooms().then(function (rooms) {
+        var result;
+        angular.forEach(rooms, function(room) {
```

Comment:
Take into account that we use [lodash](https://lodash.com/) so you can take advantage of functions like [find](https://lodash.com/docs#find). It will make your life easier :) (I've no tested this code, but something similar should work).

```javascript
result = _.find(rooms, function(room) { return room.id == roomID; });
```